rest_framework_nested/relations.py

```diff
@@ -81,14 +81,14 @@ def get_object(self, view_name, view_args, view_kwargs):
 
         return self.get_queryset().get(**kwargs)
 
-    def use_pk_only_optimization(self):
+    def use_pk_only_optimization(self) -> bool:
         return False
 
-    def to_internal_value(self, data):
+    def to_internal_value(self, data: Any) -> T_Model:
         try:
             return super().to_internal_value(data)
         except ValidationError as err:
-            if err.detail[0].code != 'no_match':
+            if err.detail[0].code != 'no_match':  # type: ignore[union-attr,index]
```

Comment:
This code is a bit iffy. `err.detail` isn't guaranteed to be a list of `ErrorDetail`. It can be an arbitrarily nested combination of `list`, `dict` and `ErrorDetail`.